scripts/dev/zero-downtime-deploy.sh

```
@@ -0,0 +1,238 @@
+    if ! sudo nginx -t 2>/dev/null; then
+        log "nginx configuration test failed, rolling back."
+        sudo cp "$backup_conf" "$nginx_conf"
+        sudo rm "$backup_conf"
+        return 1
+    fi
+
+    sudo nginx -s reload
+    log "nginx reloaded successfully"
+
+    sleep 2
+    local response=$(curl -s -o /dev/null -w "%"{http_code}"" "http://localhost/" 2>/dev/null || echo "000")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)